06.bowling_object/game.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  FINAL_FRAME_SHOT = 18
+  FINAL_FRAME = 9
+
+  def initialize(result)
+    @result = result
+  end
+
+  def calculate_total_score
+    frames = generate_frames(@result)
+    frames.each.sum { |frame| frame.calculate_score(frames) }
+  end
+
+  private
+
+  def generate_frames(result)
+    pin_counts = result.split(',')
+
+    shots = []
+    pin_counts.each do |pin_count|
+      if shots.size < FINAL_FRAME_SHOT && pin_count == 'X'
+        shots << Shot.new(pin_count)
+        shots << Shot.new('0')
```

Comment:
'X'が特別な意味を持つ、ということはShotクラスに判定させるのがいいかもしれません。
今だとこのGameクラスとShotクラスの両方が 'X' の意味を知っている状態なので。

たとえばこういうコードにしてみるのはどうでしょうか？

```suggestion
      shot = Shot.new(pin_count)
      if shots.size < FINAL_FRAME_SHOT && shot.strike?
        shots << shot
        shots << Shot.new('0')
```

Reply:
アドバイスありがとうございました！
`strike?`メソッドはShotクラスに移動し、フレーム単位ではなくショット単位でストライクの判別をするようにしました。